### ISSUE TITLE
[BUG]: NZB category mismatches & improperly syncing NZB categories

### ISSUE THREAD

**Issue by netty0** [Prowlarr/Prowlarr#587]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

**[In Prowlarr I only enabled the following for Radarr:](https://i.imgur.com/VkcZPl5.png)**
```
 - Movies        (2000)
 - Movies/HD     (2040)
 - Movies/BluRay (2050)
 - Movies/WEB-DL (2080)
```

In Radarr numerical category IDs only partially match for these NZB indexers:

**[DrunkenSlug](https://i.imgur.com/WgLTRCZ.png)** 
```
 - 3D (2050) & BluRay (2060) reversed in respect to Radarr 3D (2060) & BluRay (2050).
 - Other (2999) yet Prowlarr Other (2020)
```
**[NZBCat](https://i.imgur.com/iKBIIGS.png)**
```
- 3D (2050) & BluRay (2060) reversed in respect to Radarr 3D (2060) & BluRay (2050).
- Other (2999) yet Prowlarr Other (2020)
```
**[SimplyNZBs](https://i.imgur.com/Kh9QddF.png)**
```
- 3D (2050) & BluRay (2060) reversed in respect to Radarr 3D (2060) & BluRay (2050).
- Other (2999) yet Prowlarr Other (2020)```
```
**[NZBplanet](https://i.imgur.com/uW2y8Cu.png)**
```
- Cam (2080) reversed in respect to WEB-DL (2080) on Radarr
```
**[Tabula Rasa](https://i.imgur.com/nIUQBaj.png)**
```
- 3D (2050) & BluRay (2060) reversed in respect to Radarr 3D (2060) & BluRay (2050).
- Other (2999) yet Prowlarr Other (2020)
- X265 (2090) missing in Radarr
- Erroneously enabled Foreign (2010), SD (2030), UHD (2045), & DVD (2070) [Does not disable 
```
**[NZBfinder](https://i.imgur.com/ED4VwSG.png)**
```
- 3D (2050) & BluRay (2060) reversed in respect to Radarr 3D (2060) & BluRay (2050).
- Erroneously enabled Foreign (2010), SD (2030), UHD (2045), & DVD (2070) 
```

### Expected Behavior

Specific categories selected in Prowlarr to respectively represent the same categories that get enabled in Radarr.

### Environment

```markdown
- OS: Raspberry Pi OS Lite (2021-10-30 image)
- Prowlarr: 0.1.2.1046-ls168 by linuxserver.io
- Docker Install: ghcr.io/linuxserver/prowlarr:nightly
- Using Reverse Proxy: No
- Browser: FireFox 94.0.1
```

### What branch are you running?

Nightly

### Anything else?

- Full Sync enabled
- Prowlarr `Tasks` -> `Application Indexer Sync` selected several times to ensure sync.
- Free NZB indexers (no invite needed): NZBfinder & Tabula Rasa
- The other NZB Indexers used are [some of the largest premium NZB indexers.](https://github.com/theotherp/nzbhydra/wiki/Indexer-API-Limits)

[AB#1885](https://dev.azure.com/Servarr/7ab38f4e-5a57-4d70-84f4-94dd9bc5d6df/_workitems/edit/1885)

**Comment by neo-neo1** [comment]
Can confirm. I've experienced this also.

**Comment by bakerboy448** [comment]
Ta or Q can reopen if needed

this isn't an issue

> Movies        (2000)

this covers everything under the movies category not only 2000 itself, so it'd be expected if the main category is selected then all sub categories for each indexer will be selected as well

categories displayed in prowlarr settings are the newznab standard 

categories displayed in radarr come from the indexer capabilities  call/query

**Comment by neo-neo1** [comment]
> Ta or Q can reopen if needed
> 
> this isn't an issue
> 
> > Movies        (2000)
> 
> this covers everything under the movies category not only 2000 itself, so it'd be expected if the main category is selected then all sub categories for each indexer will be selected as well
> 
> categories displayed in prowlarr settings are the newznab standard
> 
> categories displayed in radarr come from the indexer capabilities call/query

You are grossly overlooking this issue.

**Comment by bakerboy448** [comment]
> > Ta or Q can reopen if needed
> > this isn't an issue
> > > Movies        (2000)
> > 
> > 
> > this covers everything under the movies category not only 2000 itself, so it'd be expected if the main category is selected then all sub categories for each indexer will be selected as well
> > categories displayed in prowlarr settings are the newznab standard
> > categories displayed in radarr come from the indexer capabilities call/query
> 
> You are grossly overlooking this issue. Read the post. Category IDs do NOT match up.

Enabling 2000 means use every 2XXX category.

categories do not match up because they're two different things.

Prowlarr displays only the newznab standard categories

Radarr displays the categories the indexer reports it supports

these do not necessarily match 1x1

by saying to use 2000 you're saying to enable any and all 2XXX movie categories

**Comment by netty0** [comment]
> ays only the newznab standard categories
> 
> Radarr displays the categories the indexer reports it supports
> 
> these do not necessarily match 1x1

I'm earnestly baffled. You nit pick something minuscule in a post which has nothing to do with the problem as a whole and continue to double down. 

Look at the categories ID mismatching for NZBcat, nzbplanet, etc.. You enable 3D on Prowlarr and instead BluRay will be enabled in Radarr.

**Comment by neo-neo1** [comment]
> Radarr displays the categories the indexer reports it supports
> 
> these do not necessarily match 1x1

The categories IDs for literally some of the most biggest NZB Indexers do not match up. One example out of many: If you disable BluRay on Prowlarr you will end up disabling 3D in Radarr.

**Comment by bakerboy448** [comment]
Taking NZBPlanet and 2080 for example

> - Cam (2080) reversed in respect to WEB-DL (2080) on Radarr

Their Caps calls reports
https://www.nzbplanet.net/api/?t=caps
`<subcat id="2080" name="Cam"/>`

2080 - like many of the others is not a Standard Category.
https://newznab.readthedocs.io/en/latest/misc/api/#predefined-categories

NZBFinder has this as <subcat id="2080" name="WEBDL"/>
NZBGeek does not have 2080
DS does not have 2080.
Unmentionable has this as <subcat id="2080" name="Remux"/>

Ultimately the sub-categories make no difference and again you're missing the point @neo-neo1 
If 2000 is enabled, nothing additional in 2XXX categories matter.
2000 means anything and everything in a 2XXX category.

This is something that would be handled when Indexer-App specific settings are brought into Prowlarr.

Bottom line is there is no impact if one is enabling 2000 to begin with.

**Comment by neo-neo1** [comment]
> So which category is 2080? is it Cam? Is it WebDL? Is it Remux? - Pick 1 and only 1

Perhaps manually correcting the category ID mismatch for the specific indexers rather than paint everything with the same brush? These aren't random fly-by-night indexers. If IDs don't match up for the most widely used NZB Indexers what's the point of a standard no one follows?

> 
> Ultimately the sub-categories make no difference and again you're missing the point @neo-neo1 If 2000 is enabled, nothing additional selected in 2XXX categories matter. 2000 means anything and everything in a 2XXX category.
>
> Bottom line is there is no impact if one is enabling 2000 to begin with.

a) I fully understand your point. But why keep bringing it up as if I'm arguing against it? it has nothing to do with category ID mismatches.

b) Do this experiment for for me: On Prowlarr disable only 2000 for Radarr (leave everything else default) and click the Test button. Do you get a green check-mark or red exclamation mark? Strange question, I know.

**Comment by bakerboy448** [comment]
> b) Do this experiment for for me: On Prowlarr disable only 2000 for Radarr (leave everything else default) and click the Test button. Do you get a green check-mark or red exclamation mark? Strange question, I know.

Yup that'd be a bug IMO - https://github.com/Prowlarr/Prowlarr/issues/588 - Nice Find.

> Perhaps manually correcting the category ID mismatch for the specific indexers rather than paint everything with the same brush?

A lot of work for little gain. If you have 5 different indexers say the same category number is 5 different types, then what single category do you display in Prowlarr?  No matter what, you're wrong.  Again  - this is solved by the future app/indexer settings handling.

> If IDs don't match up for the most widely used NZB Indexers what's the point of a standard no one follows?

Great question - why can't anyone follow standards? I don't know.

> it has nothing to do with category ID mismatches which is 90% of the problem at hand.

The fact 2000 covers 2XXX negates this entire issue as moot.

**Comment by neo-neo1** [comment]
> A lot of work for little gain. If you have 5 different indexers say the same category number is 5 different types, then what single category do you display in Prowlarr? 
 
 Where's the cutoff mark for what is and isn't of little gain? If the largest NZB indexers don't represent the same category IDs as Prowlarr would that not make it worth the while to cover them? Let's set the cutoff point at top 10, no? Too much?

**Comment by netty0** [comment]
> The fact 2000 covers 2XXX negates this entire issue as moot.

2000 was only included in my OP was so Prowlarr would connect. Due to bug referenced above. Pretend 2000 was never written and issue at hand still exists. Prowlarr does not accurately represent cat ids for major indexers. **Any user deviating from Prowlarr's default enabled categories will enable/disable the wrong categories. That is the core issue.**

**Comment by bakerboy448** [comment]
> Prowlarr does not accurately represent cat ids for major NZB indexers. Any user deviating from Prowlarr's default NZB categories will enable or disable the wrong categories. That is the core issue.

Backwards my friend - several major indexers fail to follow the standards that exist and choose to deviate from them.
Regardless - app-indexers settings resolves this when we get that implemented.  Anything in the interim - given the lack of standards being followed - is simply a band-aid half-fix.

**Comment by bakerboy448** [comment]
Will be fixed with the usenet switch to YML

https://github.com/Prowlarr/Prowlarr/pull/644

**Comment by bakerboy448** [comment]
Better cat parsing added w/o YML 

https://github.com/Prowlarr/Prowlarr/commit/320161e0513c1e52c78b6d0fed2f59b7541b8a19
